app/electron/main.ts

```diff
@@ -456,7 +412,80 @@ class PluginManagerEventListeners {
   }
 }
 
-function startServer(flags: string[] = []): ChildProcessWithoutNullStreams {
+/**
+ * Returns the user's preferred shell or a fallback shell.
+ * @returns A promise that resolves to the shell path.
+ */
+async function getShell(): Promise<string> {
+  // Fallback chain
+  const shells = ['/bin/zsh', '/bin/bash', '/bin/sh'];
+  let userShell = '';
+
+  try {
+    userShell = userInfo().shell || process.env.SHELL || '';
+    if (userShell) shells.unshift(userShell);
+  } catch (error) {
+    console.error('Failed to get user shell:', error);
+  }
+
+  for (const shell of shells) {
+    try {
+      await fsPromises.stat(shell);
+      return shell;
+    } catch (error) {
+      console.error(`Shell not found: ${shell}, error: ${error}`);
+    }
+  }
+
+  console.error('No valid shell found, defaulting to /bin/sh');
+  return '/bin/sh';
+}
+
+/**
+ * Retrieves the environment variables from the user's shell.
+ * @returns A promise that resolves to the shell environment.
+ */
+async function getShellEnv(): Promise<NodeJS.ProcessEnv> {
+  const execPromisify = promisify(exec);
+  const shell = await getShell();
+  const isWindows = process.platform === 'win32';
+
+  // For Windows, just return the current environment
+  if (isWindows) {
+    return { ...process.env };
+  }
+
+  // For Unix-like systems
+  const isZsh = shell.includes('zsh');
+  // interactive is supported only on zsh
+  const shellArgs = isZsh ? ['--login', '--interactive', '-c'] : ['--login', '-c'];
```

Comment:
Why not interactive here for other shells?

Reply:
![image](https://github.com/user-attachments/assets/0e86aee2-dbb7-4ce4-baed-036bb2bbe458)